app/controllers/genres_controller.rb

```diff
@@ -42,6 +42,10 @@ def destroy_empty
     Genre.left_outer_joins(:tracks).where(tracks: { id: nil }).destroy_all
   end
 
+  def merge
+    render json: @genre.errors, status: :unprocessable_entity unless @genre.merge(Genre.find(params[:old_genre_id]))
```

Comment:
The error handling here doesn't make much sense. `@genre` is never edited so will never get errors and if `@genre.merge(...)` returns something falsy it will be because the `other.destroy` statement failed.